tests/ts/query-builder-api/other-methods.ts

```diff
@@ -0,0 +1,202 @@
+    .joinRelation('children')
+    .where('age', '>', '21')
+    .debug();
+
+  const personId = 1;
+  const pets = await Person.relatedQuery('pets').for(personId);
+
+  const builder = Person.query().context({ something: 'hello' });
+  const context = builder.context();
+
+  Person.query().context({
+    runBefore(result: any, builder: any) {
+      return result;
+    },
+    runAfter(result: any, builder: any) {
+      return result;
+    },
+    onBuild(builder: any) {}
+  });
+
+  Person.query()
+    .withGraphFetched('[movies, children.movies]')
+    .context({
+      onBuild(builder: any) {
+        builder.withSchema('someSchema');
+      }
+    });
+
+  builder.mergeContext({
+    foo: 'bar'
+  });
+
+  builder.tableNameFor(Person);
+  builder.tableRefFor(Person);
+
+  builder.reject('something went wrong');
+  builder.resolve({});
+
+  builder.isExecutable();
+
+  builder.isFind();
+  builder.isInsert();
+  builder.isUpdate();
+  builder.isDelete();
+  builder.isRelate();
+  builder.isUnrelate();
+  builder.isInternal();
+
+  builder.hasWheres();
+  builder.hasSelects();
+  builder.hasWithGraph();
+  builder.has('range');
+  builder.clear('orderBy').has('orderBy');
+
+  Person.query()
+    .runBefore(async result => {
+      console.log('hello 1');
+
+      console.log('hello 2');
+      return result;
+    })
+    .runBefore(result => {
+      console.log('hello 3');
+      return result;
+    });
+
+  Person.query()
+    .onBuild(builder => {
+      builder.where('id', 1);
+    })
+    .onBuild(builder => {
+      builder.orWhere('id', 2);
+    });
+
+  Person.query().onBuildKnex((knexBuilder, objectionBuilder) => {
+    knexBuilder.where('id', 1);
+  });
+
+  Person.query()
+    .runAfter(async (models, queryBuilder) => {
+      return models;
+    })
+    .runAfter(async (models, queryBuilder) => {
+      models.push(Person.fromJson({ firstName: 'Jennifer' }));
+      return models;
+    });
+
+  Person.query()
+    .onError(async (error, queryBuilder) => {
+      if (error instanceof UniqueViolationError) {
+        return { error: 'some error occurred' };
+      } else {
+        return Promise.reject(error);
+      }
+    })
+    .where('age', '>', 30);
+
+  await Person.query()
+    .joinRelation('children.children.pets')
+    .select('children:children:pets.*')
+    .castTo(Animal);
+
+  await Person.query()
+    .joinRelation('children.pets')
+    .select(['children:pets.id as animalId', 'children.firstName as childFirstName'])
+    .castTo(Model);
+
+  const modelClass = builder.modelClass();
+
+  builder.toString();
+  builder.toSql();
+
+  Person.query()
+    .skipUndefined()
+    .where('firstName', 'something');
+
+  const trx = await transaction.start(Person);
+  builder.transacting(trx);
+
+  builder.clone();
+  builder.execute();
+
+  builder.then(
+    () => {
+      console.log('success');
+    },
+    () => {
+      console.log('error');
+    }
+  );
+
+  // builder.map((obj) => obj);
```

Comment:
These are not present in the typings currently, but I wonder if they are being deprecated like the other similar methods (`pluck`, etc).